mlir/lib/Dialect/Presburger/Parser.cpp

```diff
@@ -365,6 +471,47 @@ LogicalResult Parser::parseAnd(std::unique_ptr<Expr> &expr) {
   return success();
 }
 
+/// Parse a piece of a piecewise Presburger expression.
+///
+///  piece ::= `(`pb-sum`) : (`pb-or-expr`)`
+///
+LogicalResult
+Parser::parsePieces(SmallVector<std::unique_ptr<PieceExpr>, 4> &pieces) {
+  do {
+    lexer.next();
```

Comment:
What is this supposed to consume? The semicolon between pieces? But then why does the first run of the loop also do this? Please clarify with a comment, and mention the delimiter between pieces in the documentation at the top

Reply:
This is my bad. The calling method doesn't consume the previous token, therefore this has to be done here. I did not notice (or ignored it), as this made the implementation easier. 

Furthermore it seems that the comment is wrong. This consumes a list of pieces, not only one.

Reply:
Changed it to use similar concepts as `parseAnd` and `parseOr`